# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/patches/nfs-utils-no-getgrouplist.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- nfs-utils-1.1.0/utils/mountd/cache.c~	2007-05-10 20:40:57.000000000 -0700
+++ nfs-utils-1.1.0/utils/mountd/cache.c	2007-06-07 15:29:28.000000000 -0700
@@ -131,17 +131,17 @@
 	pw = getpwuid(uid);
 	if (!pw)
 		rv = -1;
-	else {
-		rv = getgrouplist(pw->pw_name, pw->pw_gid, groups, &ngroups);
-		if (rv == -1 && ngroups >= 100) {
-			groups = malloc(sizeof(gid_t)*ngroups);
-			if (!groups)
-				rv = -1;
-			else
-				rv = getgrouplist(pw->pw_name, pw->pw_gid,
-						  groups, &ngroups);
-		}
-	}
+//	else {
+//		rv = getgrouplist(pw->pw_name, pw->pw_gid, groups, &ngroups);
+//		if (rv == -1 && ngroups >= 100) {
+//			groups = malloc(sizeof(gid_t)*ngroups);
+//			if (!groups)
+//				rv = -1;
+//			else
+//				rv = getgrouplist(pw->pw_name, pw->pw_gid,
+//						  groups, &ngroups);
+//		}
+//	}
 	qword_printint(f, uid);
 	qword_printint(f, time(0)+30*60);
 	if (rv >= 0) {
